src/status_im2/subs/communities.cljs

```diff
@@ -209,14 +209,36 @@
         (sort-by :position)
         (into []))))
 
-(defn reduce-over-categories
-  [community-id
-   joined
+(defn- get-chat-token-permissions
+  [{community-id :id permissions :token-permissions} chat-id]
+  (let [chat-composite-id (str community-id chat-id)]
+    (->> permissions
+         (map second)
+         (filter (fn [{composite-ids :chat_ids}]
+                   (some (fn [composite-id]
+                           (= composite-id chat-composite-id))
+                         composite-ids))))))
+
+(defn- get-chat-lock-state
+  "Returns the chat lock state.
+
+  - Nil:   no lock  (there are no permissions for the chat)
+  - True:  locked   (there are permissions and can-post? is false)
+  - False: unlocked (there are permissions and can-post? is true)"
+  [community {chat-id :id can-post? :can-post?}]
+  (let [chat-permissions (get-chat-token-permissions community chat-id)]
+    (if (empty? chat-permissions)
+      nil
+      (not can-post?))))
```

Comment:
About the value returned to identify the chat lock state, I think it's not the best to mix return types that could be confused, I'm not against returning `nil`, but I think it's not the best to return `nil` when the other possible returned values are `true` and `false`.

Seems not easy to read/understand to check for `(nil? (get-chat-lock-state ,,,)`, `(true? (get-chat-lock-state ,,,)` and `(false? (get-chat-lock-state ,,,)`.

I'd suggest to return a keyword: `:no-lock`, `:locked`  and `:unlocked`.

Reply:
I agree @ulisesmac, but this is the expectation in the quo component that actually uses this argument. If I do as you suggest, which I prefer, than the view will need to do extra transformation for the component, so in the end we would end up with more code and duplication.

Ideally the component should be refactored as well to not use `nil` to mean "no lock", but I prefer to not do this in this PR if possible.